kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/SessionDoc.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.db.mongo;
+
+import com.mongodb.BasicDBObject;
+import dev.responsive.kafka.internal.utils.SessionKey;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Objects;
+import org.bson.codecs.pojo.annotations.BsonCreator;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+
+public class SessionDoc {
+
+  public static final String ID = "_id";
+  public static final String VALUE = "value";
+  public static final String EPOCH = "epoch";
+  public static final String TOMBSTONE_TS = "tombstoneTs";
+
+  // Subfields of the composite key _id
+  public static final String ID_RECORD_KEY = "key";
+  public static final String ID_SESSION_END_TS = "sessionEndTs";
+  public static final String ID_SESSION_START_TS = "sessionStartTs";
+
+  BasicDBObject id;
+  byte[] value;
+  long epoch;
+  Date tombstoneTs;
+
+  public SessionDoc() {
+  }
+
+  @BsonCreator
+  public SessionDoc(
+      @BsonProperty(ID) BasicDBObject id,
+      @BsonProperty(VALUE) byte[] value,
+      @BsonProperty(EPOCH) long epoch,
+      @BsonProperty(TOMBSTONE_TS) Date tombstoneTs
+  ) {
+    this.id = id;
+    this.value = value;
+    this.epoch = epoch;
+    this.tombstoneTs = tombstoneTs;
+  }
+
+  public BasicDBObject getKey() {
+    return id;
+  }
+
+  public void setKey(final BasicDBObject id) {
+    this.id = id;
+  }
+
+  public void setValue(final byte[] value) {
+    this.value = value;
+  }
+
+  public void setEpoch(final long epoch) {
+    this.epoch = epoch;
+  }
+
+  public byte[] getValue() {
```

Comment:
QQ: are the getter names enforced by the bson format? If not, let's drop the `get` from this to be consistent with naming elsewhere in the code

(I know we're inconsistent in some of the other Mongo Doc classes right now, there was some confusion around what was required for bson objects. And clearly we still got some things wrong)